frontend/packages/dev-console/src/components/project-access/ProjectAccess.tsx

```diff
@@ -23,21 +23,27 @@ import {
 } from './project-access-form-submit-utils';
 import { validationSchema } from './project-access-form-validation-utils';
 import ProjectAccessForm from './ProjectAccessForm';
-import { Verb, UserRoleBinding, Roles, roleBinding } from './project-access-form-utils-types';
+import { Verb, UserRoleBinding, roleBinding } from './project-access-form-utils-types';
 
 export interface ProjectAccessProps {
   formName: string;
   namespace: string;
   roleBindings?: { data: []; loaded: boolean; loadError: {} };
+  roles: Roles;
 }
 
-const ProjectAccess: React.FC<ProjectAccessProps> = ({ formName, namespace, roleBindings }) => {
+const ProjectAccess: React.FC<ProjectAccessProps> = ({
+  formName,
+  namespace,
+  roleBindings,
+  roles,
+}) => {
   const { t } = useTranslation();
-  if (!roleBindings.loaded && _.isEmpty(roleBindings.loadError)) {
+  if ((!roleBindings.loaded && _.isEmpty(roleBindings.loadError)) || _.isEmpty(roles)) {
```

Comment:
With the changed hook and changed props above, you can check for the loaded state, instead of expecting that the data are not empty.

```suggestion
  if ((!roleBindings.loaded && _.isEmpty(roleBindings.loadError)) || !roles.loaded) {
```